services/lumberjack/src/service.js

```diff
@@ -0,0 +1,293 @@
+    this._telemInterval = options.telemInterval;
+    this._gTimes = options.gTimes;
+    this._dbName = options.dbName;
+    this._influx = null;
+  }
+
+  /** Start the service. */
+  async start() {
+    logger.debug('Starting service.');
+
+    /** Database configuration */
+    this._influx = new InfluxDB({
+      host: this._influxHost,
+      port: this._influxPort,
+      database: this._dbName,
+      schema: [
+        {
+          measurement: 'ping',
+          fields: {
+            apiPing: FieldType.INTEGER,
+            devicePing: FieldType.INTEGER
+          },
+          tags: [
+            'host',
+            'port',
+            'name'
+          ]
+        },
+        {
+          measurement: 'upload-rate',
+          fields: {
+            total_1: FieldType.INTEGER,
+            total_5: FieldType.INTEGER,
+            fresh_1: FieldType.INTEGER,
+            fresh_5: FieldType.INTEGER
+          },
+          tags: [
+            'host',
+            'port'
+          ]
+        },
+        {
+          measurement: 'telemetry',
+          fields: {
+            gstatus: FieldType.INTEGER,
+            pstatus: FieldType.INTEGER
+          },
+          tags: [
+            'host',
+            'port'
+          ]
+        }
+      ]
+    });
+    // Create database if it doesn't exist
+    const names = await this._influx.getDatabaseNames();
+    if (!names.includes(this._dbName))
+      await this._influx.createDatabase(this._dbName);
+
+    this._startTasks();
+    logger.debug('Service started');
+  }
+
+  /** Stop the service. */
+  async stop() {
+    logger.debug('Stopping service.');
+
+    await Promise.all([
+      this._pingTask.stop(),
+      this._uploadRateTask.stop(),
+      this._groundTelemetryTask.stop(),
+      this._planeTelemetryTask.stop(),
+    ]);
+
+    logger.debug('Service stopped.');
+  }
+
+  _startTasks() {
+    this._pingTask =
+      createTimeoutTask(this._ping.bind(this),
+        this._pingInterval)
+        .on('error', logger.error)
+        .start();
+    this._uploadRateTask =
+      createTimeoutTask(this._uploadRate.bind(this),
+        this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+    this._groundTelemetryTask =
+      createTimeoutTask(this._groundTelemetry.bind(this),
+        this._telemInterval)
+        .on('error', logger.error)
+        .start();
+    this._planeTelemetryTask =
+      createTimeoutTask(this._planeTelemetry.bind(this),
+        this._telemInterval)
+        .on('error', logger.error)
+        .start();
+  }
+
+  /** Get service ping data and write to the database */
+  async _ping() {
+    // Get ping data
+    let ping;
+    try {
+      ping =
+        (await request.get('http://' + this._pingHost + ':' +
+          this._pingPort + '/api/ping')
+          .proto(stats.PingTimes)
+          .timeout(1000)).body;
+    } catch (err) {
+      logger.error(err);
+      await this._influx.writeMeasurement('ping', [
```

Comment:
So here I don't think you need to make a write to the database at all, and instead just have the function return.